modules/codegen/src/main/scala/com/twilio/guardrail/generators/Scala/CirceProtocolGenerator.scala

```diff
@@ -461,7 +522,28 @@ object CirceProtocolGenerator {
           q"implicit val guardrailEncodeLocalDateTime: Encoder[LocalDateTime] = Encoder[LocalDateTime]",
           q"implicit val guardrailEncodeLocalTime: Encoder[LocalTime] = Encoder[LocalTime]",
           q"implicit val guardrailEncodeOffsetDateTime: Encoder[OffsetDateTime] = Encoder[OffsetDateTime]",
-          q"implicit val guardrailEncodeZonedDateTime: Encoder[ZonedDateTime] = Encoder[ZonedDateTime]"
+          q"implicit val guardrailEncodeZonedDateTime: Encoder[ZonedDateTime] = Encoder[ZonedDateTime]",
+          q"""sealed trait Property[+T] {
+               def fold[R](ifAbsent: => R,
```

Comment:
i think having to `toOption` would be a useful method here - i suspect his is what people will often want to do

Reply:
Replaced `flatten` with `toOption`. Also added `Property.fromOption`.